common/src/main/cpp/extensions/openxr_fb_body_tracking_extension_wrapper.cpp

```diff
@@ -0,0 +1,338 @@
+
+#include "extensions/openxr_fb_body_tracking_extension_wrapper.h"
+
+#include <godot_cpp/classes/object.hpp>
+#include <godot_cpp/classes/open_xrapi_extension.hpp>
+#include <godot_cpp/classes/xr_server.hpp>
+#include <godot_cpp/variant/utility_functions.hpp>
+
+using namespace godot;
+
+/// Joint mapping table entry
+struct JointMapEntry {
+	/// Joint in Godot XRBodyTracker
+	XRBodyTracker::Joint xr_joint;
+
+	/// Joint in OpenXR XrBodyJointFB
+	XrBodyJointFB fb_joint;
+
+	/// Joint rotation
+	Quaternion rotation;
+};
+
+/// Joint mapping table
+static const JointMapEntry joint_table[] = {
+	// Root joint
+	{ XRBodyTracker::JOINT_ROOT, XR_BODY_JOINT_ROOT_FB, Quaternion(0.0, 0.0, 0.0, 1.0) },
+
+	// Upper body joints
+	{ XRBodyTracker::JOINT_HIPS, XR_BODY_JOINT_HIPS_FB, Quaternion(-0.5, 0.5, 0.5, 0.5) },
+	{ XRBodyTracker::JOINT_SPINE, XR_BODY_JOINT_SPINE_LOWER_FB, Quaternion(-0.5, 0.5, 0.5, 0.5) },
+	{ XRBodyTracker::JOINT_CHEST, XR_BODY_JOINT_SPINE_UPPER_FB, Quaternion(-0.5, 0.5, 0.5, 0.5) },
+	{ XRBodyTracker::JOINT_UPPER_CHEST, XR_BODY_JOINT_CHEST_FB, Quaternion(-0.5, 0.5, 0.5, 0.5) },
+	{ XRBodyTracker::JOINT_NECK, XR_BODY_JOINT_NECK_FB, Quaternion(-0.5, 0.5, 0.5, 0.5) },
+	{ XRBodyTracker::JOINT_HEAD, XR_BODY_JOINT_HEAD_FB, Quaternion(-0.5, 0.5, 0.5, 0.5) },
+	{ XRBodyTracker::JOINT_LEFT_SHOULDER, XR_BODY_JOINT_LEFT_SHOULDER_FB, Quaternion(0.0, 0.0, 0.7071067811865475244, 0.7071067811865475244) },
+	{ XRBodyTracker::JOINT_LEFT_UPPER_ARM, XR_BODY_JOINT_LEFT_ARM_UPPER_FB, Quaternion(-0.7071067811865475244, 0.7071067811865475244, 0.0, 0.0) },
+	{ XRBodyTracker::JOINT_LEFT_LOWER_ARM, XR_BODY_JOINT_LEFT_ARM_LOWER_FB, Quaternion(0.5, -0.5, -0.5, -0.5) },
+	{ XRBodyTracker::JOINT_RIGHT_SHOULDER, XR_BODY_JOINT_RIGHT_SHOULDER_FB, Quaternion(0.7071067811865475244, 0.7071067811865475244, 0.0, 0.0) },
+	{ XRBodyTracker::JOINT_RIGHT_UPPER_ARM, XR_BODY_JOINT_RIGHT_ARM_UPPER_FB, Quaternion(0.0, 0.0, -0.7071067811865475244, 0.7071067811865475244) },
+	{ XRBodyTracker::JOINT_RIGHT_LOWER_ARM, XR_BODY_JOINT_RIGHT_ARM_LOWER_FB, Quaternion(0.5, 0.5, -0.5, 0.5) },
+
+	// Left hand joints
+	{ XRBodyTracker::JOINT_LEFT_HAND, XR_BODY_JOINT_LEFT_HAND_WRIST_FB, Quaternion(0.5, -0.5, -0.5, -0.5) },
+	{ XRBodyTracker::JOINT_LEFT_PALM, XR_BODY_JOINT_LEFT_HAND_PALM_FB, Quaternion(0.5, -0.5, -0.5, -0.5) },
+	{ XRBodyTracker::JOINT_LEFT_WRIST, XR_BODY_JOINT_LEFT_HAND_WRIST_FB, Quaternion(0.5, -0.5, -0.5, -0.5) },
+	{ XRBodyTracker::JOINT_LEFT_THUMB_METACARPAL, XR_BODY_JOINT_LEFT_HAND_THUMB_METACARPAL_FB, Quaternion(0.5, -0.5, -0.5, -0.5) },
+	{ XRBodyTracker::JOINT_LEFT_THUMB_PHALANX_PROXIMAL, XR_BODY_JOINT_LEFT_HAND_THUMB_PROXIMAL_FB, Quaternion(0.5, -0.5, -0.5, -0.5) },
+	{ XRBodyTracker::JOINT_LEFT_THUMB_PHALANX_DISTAL, XR_BODY_JOINT_LEFT_HAND_THUMB_DISTAL_FB, Quaternion(0.5, -0.5, -0.5, -0.5) },
+	{ XRBodyTracker::JOINT_LEFT_THUMB_TIP, XR_BODY_JOINT_LEFT_HAND_THUMB_DISTAL_FB, Quaternion(0.5, -0.5, -0.5, -0.5) },
```

Comment:
This should be `XR_BODY_JOINT_LEFT_HAND_THUMB_TIP_FB`

Reply:
Nice catch - I flubbed that one populating the table.